src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp

```diff
@@ -1445,44 +1451,26 @@ void AlgorithmPaintToolBox::addSliceToStack(medAbstractView * view,const unsigne
             m_redoStacks->value(view)->clear();
 }
 
-void AlgorithmPaintToolBox::closeView()
+void AlgorithmPaintToolBox::clear()
 {
-    // close every data in the view
-    medAbstractView* view = qobject_cast<medAbstractView*>(QObject::sender());
-    medAbstractLayeredView* layeredView = dynamic_cast<medAbstractLayeredView*>(view);
-
-    if (layeredView->layersCount() > 0)
+    if (currentView && (currentView->layersCount()>0))
     {
-        // remove a data, which will recursively call this closeView() function
-        // until there is no more data in the view
-        layeredView->removeLayer(0);
-    }
-    else
-    {
-        // clean the toolbox when there is no data left in the view
-        onViewClosed(view);
-    }
-}
-
-void AlgorithmPaintToolBox::onViewClosed(medAbstractView* viewClosed)
-{
-    clearMask();
-
-    if (m_undoStacks->value(viewClosed))
-    {
-        m_undoStacks->value(viewClosed)->clear();
-        m_redoStacks->value(viewClosed)->clear();
-        m_undoStacks->remove(viewClosed);
-        m_redoStacks->remove(viewClosed);
+        // remove the "no name" mask data in the view
+        for(int unsigned i=0; i<currentView->layersCount(); i++)
+        {
+            if (currentView->layerData(i)->identifier() == "medImageMaskAnnotationData")
+            {
+                currentView->removeLayer(i); // call clear() again
+                return;
```

Comment:
This means that the 2 lines will be called at the end. To be clear, here is the process:
 * user clicks on Clear Mask.
 * `clear()` is called.
 * a `medImageMaskAnnotationData` dataset is found and removed. End of the function.
 * at the end of the removal action, `clear()` is automatically called again.
 * no `medImageMaskAnnotationData` is found, the 2 lines are called.

:)